core/templates/pages/preferences-page/preferences-page.component.ts

```diff
@@ -176,18 +183,45 @@ export class PreferencesPageComponent {
     }
   }
 
+  private saveProfileImage(newProfilePictureDataUrl: string): void {
+    if (AppConstants.EMULATOR_MODE) {
+      this._saveProfileImageToLocalStorage(newProfilePictureDataUrl);
+    } else {
+      this._postProfileImageToServer(newProfilePictureDataUrl);
+    }
+  }
+
+  private _saveProfileImageToLocalStorage(image: string): void {
+    const newImageFile = (
+      this.imageUploadHelperService.convertImageDataToImageFile(image));
+    const reader = new FileReader();
+    reader.onload = () => {
+      const imageData = reader.result as string;
+      this.imageLocalStorageService.saveImage(
+        this.username + '_profile_picture.png', imageData);
+    };
+    reader.readAsDataURL(newImageFile);
+    // The reload is needed in order to update the profile picture
+    // in the top-right corner.
+    this.windowRef.nativeWindow.location.reload();
```

Comment:
Is it needed for local storage too? I think we should still preserve the reload but the comment might be different.

Reply:
Yep! We also want to update the profile picture on top navbar we have. Should I replace the `top-right` with `top-navbar`? I'm not sure can you suggest something?

Reply:
Not really. I just thought that for the local storage, the image will get updated, but I now realized that it is not the case.